torch/nn/modules/module.py

```diff
@@ -374,6 +393,11 @@ def to(self, *args, **kwargs):
 
         """
 
+        warnings.warn("The current behavior of `model.to()` is incorrect because it changes \
```

Comment:
This message is quite confusing because it's not obvious which definitions you are using.

1) What is incorrect?  According to whom?
2) 'in-place' vs creating a new model is pretty confusing.  Does in-place mean the parameter change in-place or just that the model is changed in-place by moving the parameters?

Reply:
I also think we should restrict warnings to just the situation that will break:
1) using cpu/cuda with force set
2) use_count greater than 1.

Reply:
You don't need to do that in this PR though, that can be separate, for this PR just introduce the flag in order to unbreak XLA.